docs/Hooks.md

```diff
@@ -293,6 +294,32 @@ fastify.addHook('onRoute', (routeOptions) => {
   routeOptions.prefix
 })
 ```
+<a name="on-encapsulation"></a>
+**'onEncapsulation'**<br>
+Triggered when Fastify runs its internal logic that enables the encapsulation.<br/>
+This hook can be useful if you are developing a plugin that needs to use the encapsulation functionality of Fastify.
```

Comment:
After reading both lines I still have no clue why it exists. I think what you're trying to say is:

> This hook is executed when a new [`plugin`](https://github.com/fastify/fastify/blob/master/docs/Plugins.md#plugins) function is registered, hence a new encapsulation context was created. This hook can be useful if you are developing a plugin that needs to know when a plugin context is created and to operate in that specific context.

I also vote for a different name e.g `onPlugin`

Reply:
I'll use your description, is quite effective :)

`onPlugin` might work, @fastify/fastify what do you think?